java/src/main/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredential.java

```diff
@@ -0,0 +1,52 @@
+package com.jongallant.azure.identity.extensions;
+
+import com.microsoft.azure.servicebus.security.SecurityToken;
+import com.microsoft.azure.servicebus.security.SecurityTokenType;
+import com.microsoft.azure.servicebus.security.TokenProvider;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.DefaultAzureCredential;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import reactor.core.publisher.Mono;
+
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A Service Bus {@link TokenProvider} that uses {@link DefaultAzureCredential} that is available in com
+ * .azure:azure-identity module. This class provides a convenient mechanism to authenticate service bus using the latest
+ * Azure Identity SDK.
+ */
+public class DefaultAzureServiceBusCredential extends TokenProvider {
+
+  private static final String SERVICEBUS_SCOPE = "https://servicebus.azure.net/.default";
+  private final DefaultAzureCredential defaultAzureCredential;
+
+  /**
+   * Creates an instance of DefaultAzureServiceBusCredential.
+   */
+  public DefaultAzureServiceBusCredential() {
+    this(new DefaultAzureCredentialBuilder().build());
+  }
+
+  /**
+   * Creates an instance of DefaultAzureServiceBusCredential using the provided {@link DefaultAzureCredential}.
+   *
+   * @param defaultAzureCredential The {@link DefaultAzureCredential} to use.
+   */
+  public DefaultAzureServiceBusCredential(DefaultAzureCredential defaultAzureCredential) {
+    this.defaultAzureCredential = defaultAzureCredential;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public CompletableFuture<SecurityToken> getSecurityTokenAsync(String audience) {
+    TokenRequestContext tokenRequestContext = new TokenRequestContext().addScopes(SERVICEBUS_SCOPE);
```

Comment:
Once you implement the others Mgmt and Fluent, you'll see how I cache in-memory.  Can you also implement that here?  I think once you implement those other two flavors you'll get a better idea of what can be shared.

Reply:
Added caching